source/tests/unit/ni_fake_service_tests.cpp

```diff
@@ -1415,6 +1415,32 @@ TEST(NiFakeServiceTests, NiFakeService_GetAnIviDanceWithATwistArray_CallsGetAnIv
     EXPECT_EQ(response.actual_size(), expected_size);
 }
 
+TEST(NiFakeServiceTests, NiFakeService_GetArrayViUInt8WithEnum_CallsGetArrayViUInt8WithEnum)
+{
+  nidevice_grpc::SessionRepository session_repository;
+  std::uint32_t session_id = create_session(session_repository, kTestViSession);
+  NiFakeMockLibrary library;
+  nifake_grpc::NiFakeService service(&library, &session_repository);
+  int array_len = 3;
+  // using string because in grpc, ViUInt8 array gets converted to string
+  std::string uint8_array = {nifake_grpc::Color::COLOR_RED, nifake_grpc::Color::COLOR_BLACK, nifake_grpc::Color::COLOR_BLUE};
```

Comment:
The library call returns raw values (not enum values). To mock that accurately, you should be using underlying values of COLOR_RED, COLOR_BLACK and COLOR_BLUE here. And then use this variable during `SetArrayArgument` on following line.

Reply:
made this change